app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ingredients_analysis/IngredientsWithTagDialogFragment.java

```diff
@@ -0,0 +1,262 @@
+                            iconResId = R.drawable.ic_monkey_unhappy;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_sad);
+                            break;
+                        case "no":
+                            titleResId = R.string.not_from_palm_oil;
+                            iconResId = R.drawable.ic_monkey_happy;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_happy);
+                            messageStr = getString(R.string.this_product_is, getString(titleResId).toLowerCase());
+                            break;
+                        default:
+                            titleResId = R.string.unknown_palm_oil;
+                            iconResId = R.drawable.ic_monkey_uncertain;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_unknown);
+                            messageStr = getString(R.string.unknown_status_missing_ingredients, getString(nameId).toLowerCase());
+                            showHelpExtract = true;
+                    }
+                    break;
+                case "vegetarian":
+                    iconResId = R.drawable.ic_egg;
+                    nameId = R.string.vegetarian;
+                    rbTextResId = R.string.preference_display_vegetarian_status;
+                    prefKey = "enableVegetarianStatusDisplay";
+                    switch (value) {
+                        case "maybe":
+                            titleResId = R.string.maybe_vegetarian;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_uncertain);
+                            break;
+                        case "unknown":
+                            titleResId = R.string.unknown_vegetarian;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_unknown);
+                            messageStr = getString(R.string.unknown_status_no_translation, getString(nameId).toLowerCase());
+                            showHelpTranslate = true;
+                            break;
+                        case "no":
+                            titleResId = R.string.non_vegetarian;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_sad);
+                            break;
+                        case "yes":
+                            titleResId = R.string.vegetarian;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_happy);
+                            messageStr = getString(R.string.this_product_is, getString(titleResId).toLowerCase());
+                            break;
+                        default:
+                            titleResId = R.string.unknown_vegetarian;
+                            iconColor = ContextCompat.getColor(getActivity(), R.color.monkey_unknown);
```

Comment:
I would extract those colors as an array or something, and give them a name that's more generic
            R.color.monkey_uncertain
            R.color.monkey_unknown
            R.color.monkey_sad
            R.color.monkey_happy

            R.color.detection_uncertain
            R.color.detection_unknown
            R.color.detection_green
            R.color.detection_red